controllers/subnamespace_controller.go

```diff
@@ -34,18 +36,105 @@ type SubnamespaceReconciler struct {
 	Scheme *runtime.Scheme
 }
 
-// +kubebuilder:rbac:groups=dana.github.com/Dana-Team/SNS,resources=subnamespaces,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=dana.github.com/Dana-Team/SNS,resources=subnamespaces/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=dana.dana.hns.io,resources=subnamespaces,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=dana.dana.hns.io,resources=subnamespaces/status,verbs=get;update;patch
+
+func (r *SubnamespaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("subspace", req.NamespacedName)
+	var subspace danav1alpha1.Subnamespace
+	var ownerNamespace v1.Namespace
+	var childNamespace v1.Namespace
+	var childNamespaceName string
+
+	// Getting reconciled subspace
+	if err := r.Get(ctx, req.NamespacedName, &subspace); err != nil {
+		log.Error(err, "Could not find Subspace")
+		return ctrl.Result{}, err
+	}
+	// Getting subspace's namespace - described as ownerNamespace
+	if err := r.Get(ctx, client.ObjectKey{Namespace: "", Name: subspace.GetNamespace()}, &ownerNamespace); err != nil {
+		log.Error(err, "Could not find owner Namespace ")
+		return ctrl.Result{}, err
```

Comment:
is this necessarily an error when a namespace is not found?

Reply:
Yes, It means that the Subspace doesn't have a Namespace which doesn't make sense.